### PR TITLE
feat(curriculum): add numbers to modules and fixes

### PR DESCRIPTION
**Pull request by fiji-flo**
**This depends on https://github.com/mdn/curriculum/pull/50**

## Summary

This resolves:
- MP-948 Supporting numbers in the titles
- MP-949 Make sure to use modules instead of module
- MP-971 Add the group label on module pages
- #10734 and #10719 Making the whole module card a link
- Steamline one of the dark background colors

---

### Before/After

|Before|After|
|------|-----|
|image|image|
|image|image|

---

## How did you test this change?

Locally

### CODE REVIEW

File: build/curriculum.ts [hunk 1]
```diff
@@ -220,11 +221,12 @@ async function readCurriculumPage(
 
     sidebar = await buildCurriculumSidebar();
     parents = await buildParents(url);
-  } else {
-    title = title
-      .replace(/^\d+\s+/, "") // Strip number prefix.
-      .replace(/ modules$/, "") // Strip "modules" suffix.
-      .replace(/Extension \d+:/, ""); // Strip "Extension" prefix.
+    if (parents.length > 1) {
+      const title = parents[parents.length - 2]?.title;
```
Comment (caugner) [inline build/curriculum.ts:225]:
```suggestion
      const title = parents.at(-2)?.title;
```

File: client/src/curriculum/module.scss [hunk 2]
```diff
@@ -21,7 +29,7 @@
         }
       }
 
-      p {
+      p.module-topic {
```
Comment (caugner) [inline client/src/curriculum/module.scss:32]: Nit: Do we need to be this explicit?

```suggestion
      .module-topic {
```

File: client/src/curriculum/module.scss [hunk 3]
```diff
@@ -32,6 +40,20 @@
           content: "Category: ";
         }
       }
+
+      p.module-group {
```
Comment (caugner) [inline client/src/curriculum/module.scss:44]: Nit: dito

```suggestion
      .module-group {
```

File: client/src/curriculum/modules-list.tsx [hunk 4]
```diff
@@ -29,7 +29,9 @@ export function ModulesListList({
               checked={i === tab}
               onChange={() => setTab(i)}
             />
-            <label htmlFor={`module-${i}`}>{modulesList.title}</label>
+            <label htmlFor={`module-${i}`}>
+              {modulesList.title.replace(/ modules$/, "")}
```
Comment (caugner) [inline client/src/curriculum/modules-list.tsx:33]: Should we have `title` and `short_title`, instead of doing these replacements in the view?

**Review by caugner — APPROVED** [review]

File: client/src/curriculum/module.scss (hunk 2 above)
Comment (fiji-flo) [reply client/src/curriculum/module.scss:32]: No, but we can. That's a discussion for having CSS guidelines in the future.

File: client/src/curriculum/modules-list.tsx (hunk 4 above)
Comment (fiji-flo) [reply client/src/curriculum/modules-list.tsx:33]: No, that's just for the representation in this specific place for now.

**Pull request merged**